### PR TITLE
Ft  refactor code

### PR DESCRIPTION
**Pull request by Bateyjosue**
# FEATURE DESCRIPTION
## TASK

In this project, we use newly acquired knowledge to refactor our previous code.

1. Follow SOLID principles.

- [ ] Pay special attention to the "Single-responsibility Principle".
- [ ] Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- [ ] Extract some pieces of logic into a separate class(es).

### Cheers

# Happy Review

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi  @Bateyjosue and @AlierPM, Kindly use the updated version  of the microverse [readme](https://github.com/microverseinc/readme-template) template for your readme, this is a requirement to get a review

**Comment by prolajumokeoni** [comment]
Hi @Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the updated version  of the microverse [readme](https://github.com/microverseinc/readme-template) template 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      @rep = gets.chomp
+      case @rep.to_i
+      when 1
+        list_books(@books)
+      when 2
+        list_person(@persons)
+      when 3
+        create_person
+      when 4
+        add_book
```
Comment (Davidosky007) [inline app.rb]: * Kindly Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit), so I strongly suggest you extract these pieces of code into your main.rb file as you are to [Pay special attention to the "Single-responsibility Principle" ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md#:~:text=Pay%20special%20attention%20to%20the%20%22Single%2Dresponsiblity%20Principle%22)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Bateyjosue @AlierPM 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well-written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @Bateyjosue 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**